extensions/podman/src/podman-install.ts

```diff
@@ -294,3 +310,97 @@ class WinInstaller extends BaseInstaller {
     });
   }
 }
+
+abstract class BaseCheck implements extensionApi.InstallCheck {
+  abstract title: string;
+  abstract execute(): Promise<extensionApi.CheckResult>;
+
+  protected createFailureResult(description?: string): extensionApi.CheckResult {
+    return { successful: false, description };
+  }
+
+  protected createSuccessfulResult(): extensionApi.CheckResult {
+    return { successful: true };
+  }
+}
+
+class WinBitCheck extends BaseCheck {
+  title = 'Windows x64';
+  private ARCH = 'x64';
+  async execute(): Promise<extensionApi.CheckResult> {
+    const currentArch = process.arch;
+    if (this.ARCH === currentArch) {
+      return this.createSuccessfulResult();
+    } else {
+      return this.createFailureResult('WSL2 works only on x64 OS.');
+    }
+  }
+}
+
+class WinVersionCheck extends BaseCheck {
+  title = 'Windows Version';
+
+  private MIN_BUILD = 18362;
+  async execute(): Promise<extensionApi.CheckResult> {
+    const winRelease = os.release();
+    if (winRelease.startsWith('10.0.')) {
+      const splitRelease = winRelease.split('.');
+      const winBuild = splitRelease[2];
+      if (Number.parseInt(winBuild) >= this.MIN_BUILD) {
+        return { successful: true };
+      } else {
+        return this.createFailureResult(
+          'To be able to run WSL2 you need Windows 10 Build 18362 or later. Please update your OS.',
+        );
+      }
+    } else {
+      return this.createFailureResult('WSL2 works only on Windows 10 and newest OS');
+    }
+  }
+}
+
+class WinMemoryCheck extends BaseCheck {
+  title = 'RAM';
+  private REQUIRED_MEM = 6 * 1024 * 1024 * 1024; // 6Gb
+
+  async execute(): Promise<extensionApi.CheckResult> {
+    const totalMem = os.totalmem();
+    if (this.REQUIRED_MEM <= totalMem) {
+      return this.createSuccessfulResult();
+    } else {
+      return this.createFailureResult('You need at least 6Gb to run Podman.');
+    }
+  }
+}
+
+class HyperVCheck extends BaseCheck {
+  title = 'Hyper-V Enabled';
+
+  async execute(): Promise<extensionApi.CheckResult> {
+    try {
+      const res = await execPromise('powershell.exe', ['(Get-Service vmcompute).DisplayName']);
+      if (res === 'Hyper-V Host Compute Service') {
+        return this.createSuccessfulResult();
+      }
+    } catch (err) {
+      // ignore error, this means that hyper-v not enabled
+    }
+    return this.createFailureResult('Hyper-V should be enabled to be able to run Podman');
```

Comment:
while it's nice that we don't do the install and continue, it could be helpful if we have a link on how to do that